src/RageSoundMixBuffer.cpp

```diff
@@ -82,7 +82,7 @@ void RageSoundMixBuffer::read( std::int16_t *pBuf )
 	{
 		float iOut = m_pMixbuf[iPos];
 		iOut = clamp( iOut, -1.0f, +1.0f );
-		pBuf[iPos] = std::lrint(iOut * 32767);
+		pBuf[iPos] = static_cast<int>(iOut * 32767);
```

Comment:
For better closeness, let's change all the `std::lrint` calls in this PR to `static_cast<int>(foo + 0.5)`, or potentially even `static_cast<int64_t>` (depending on the data type) like we did below.

Reply:
Sounds good, I like static_casting to int64_t

Reply:
Well it depends on the data type. Like here `pBuf` is a `int16_t`. So take care to cast it correctly and not just `int64_t` willy nilly.

Reply:
Btw, in this function, `pBuf` is an int16_t. I would like to expand this to a larger value as well, but I'd need to check that nothing breaks if it's bigger. (in RageSoundReader_ThreadedBuffer.cpp, you can see at the top of the file how tiny a chunk of audio is read at a time. Other things may break when this is changed, so it should be tested well before committing that change).

Reply:
Marking this resolved, but I am making a note to myself to look into ThreadedBuffer more if audio buffer stuff isn't sorted out by the existing fixes.